lib/action_controller/serialization.rb

```diff
@@ -56,10 +56,15 @@ def use_adapter?
 
     [:_render_option_json, :_render_with_renderer_json].each do |renderer_method|
       define_method renderer_method do |resource, options|
-        options.fetch(:serialization_context) do
-          options[:serialization_context] = ActiveModelSerializers::SerializationContext.new(request, options)
+        options.freeze
+        serialization_options = options.deep_dup
```

Comment:
I'm thinking not

On Wed, Mar 30, 2016 at 7:14 AM Yohan Robert <notifications@github.com>
wrote:

> In lib/action_controller/serialization.rb
> <https://github.com/rails-api/active_model_serializers/pull/1572#discussion_r57877815>
> :
>
> > @@ -56,10 +56,15 @@ def use_adapter?
> >
> >      [:_render_option_json, :_render_with_renderer_json].each do |renderer_method|
> >        define_method renderer_method do |resource, options|
> > -        options.fetch(:serialization_context) do
> > -          options[:serialization_context] = ActiveModelSerializers::SerializationContext.new(request, options)
> > +        options.freeze
> > +        serialization_options = options.deep_dup
>
> Do we actually need to dup and freeze the options at all?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rails-api/active_model_serializers/pull/1572/files/094af453fe0eb9e8978f6aeec65ec87e22b59a28#r57877815>
>